src-graphql/client.js

```diff
@@ -24,11 +24,11 @@ function fetchAllProductResources(product, client) {
   const images = product.images;
   const variants = product.variants;
 
-  if (images && images[images.length - 1].hasNextPage.valueOf()) {
+  if (images && images.length && images[images.length - 1].hasNextPage.valueOf()) {
```

Comment:
Yeah I tried that myself and it didn't work either :(

In this case I just mean that a function like `hasNextPage` shouldn't return a primitive `Boolean` *or* a `Scalar`. So I think the solution here is for the client to call `connection.pageInfo.hasNextPage.valueOf()`. Then the return type is consistent.